client/src/app/pages/questionnaires/Questionnaires.tsx

```diff
@@ -0,0 +1,480 @@
+    pushNotification({
+      title: t("toastr.success.deletedWhat", {
+        what: name,
+        type: t("terms.questionnaire"),
+      }),
+      variant: "success",
+    });
+  };
+
+  const { mutationFn: deleteQuestionnaire } = useDeleteQuestionnaireMutation(
+    onDeleteQuestionnaireSuccess,
+    onSaveQuestionnaireError
+  );
+
+  const [isImportModal, setIsImportModal] = React.useState<boolean>(false);
+  const [isDownloadTemplateModal, setIsDownloadTemplateModal] =
+    React.useState<boolean>(false);
+
+  const [isKebabOpen, setIsKebabOpen] = React.useState<number | null>(null);
+  const [questionnaireToExport, setQuestionnaireToExport] = React.useState<
+    number | null
+  >(null);
+  const [questionnaireToDelete, setQuestionnaireToDelete] = React.useState<
+    number | null
+  >(null);
+
+  const tableControls = useLocalTableControls({
+    idProperty: "id",
+    items: questionnaires,
+    columnNames: {
+      required: "Required",
+      name: "Name",
+      questions: "Questions",
+      rating: "Rating",
+      dateImported: "Date imported",
+    },
+    isSelectable: false,
+    expandableVariant: null,
+    hasActionsColumn: true,
+    filterCategories: [
+      {
+        key: "name",
+        title: t("terms.name"),
+        type: FilterType.search,
+        placeholderText:
+          t("actions.filterBy", {
+            what: t("terms.name").toLowerCase(),
+          }) + "...",
+        getItemValue: (item) => {
+          return item?.name || "";
+        },
+      },
+    ],
+    sortableColumns: ["name", "dateImported"],
+    getSortValues: (assessment) => ({
+      name: assessment.name || "",
+      dateImported: assessment.dateImported || "",
+    }),
+    initialSort: { columnKey: "name", direction: "asc" },
+    hasPagination: true,
+    isLoading: isFetching,
+  });
+  const {
+    currentPageItems,
+    numRenderedColumns,
+    selectionState: { selectedItems },
+    propHelpers: {
+      toolbarProps,
+      toolbarBulkSelectorProps,
+      filterToolbarProps,
+      paginationToolbarItemProps,
+      paginationProps,
+      tableProps,
+      getThProps,
+      getTdProps,
+    },
+  } = tableControls;
+
+  return (
+    <>
+      <PageSection variant={PageSectionVariants.light}>
+        <TextContent>
+          <Text component="h1">{t("terms.assessment")}</Text>
+        </TextContent>
+      </PageSection>
+      <PageSection>
+        <ConditionalRender
+          when={isFetching && !(questionnaires || fetchError)}
+          then={<AppPlaceholder />}
+        >
+          <div
+            style={{
+              backgroundColor: "var(--pf-v5-global--BackgroundColor--100)",
+            }}
+          >
+            <Toolbar {...toolbarProps}>
+              <ToolbarContent>
+                <ToolbarBulkSelector {...toolbarBulkSelectorProps} />
+                <FilterToolbar {...filterToolbarProps} />
+                <ToolbarGroup variant="button-group">
+                  {/* <RBAC
```

Comment:
Side-note: are we going to need to revisit RBAC on all these new views before 7.0 feature freeze? I think we have some commented-out or omitted RBAC stuff on Issues/Dependencies etc as well. Not sure if we're tracking that anywhere

Reply:
@mturley, I don't know.
@ibolton336, what do you think ?